### PR TITLE
Correct behaviour of immersive interactives

### PR DESCRIPTION
**Pull request by OliverJAsh**
In #8570, @phamann added the ability to _hide all UI_ given the display hint of immersive. We actually mean to only hide content furniture.

Before:
image

After:
image

Should we also hide the date + share buttons?

/cc @rich-nguyen @andymason @fcage @Pearson82

### CODE REVIEW

File: common/app/model/content.scala [hunk 1]
```diff
@@ -745,9 +745,10 @@ class Interactive(content: ApiContentWithMeta) extends Content(content) {
   override lazy val analyticsName = s"GFE:$section:$contentType:${id.substring(id.lastIndexOf("/") + 1)}"
 
   override lazy val metaData: Map[String, JsValue] = super.metaData + ("contentType" -> JsString(contentType))
-  override lazy val hideUi: Boolean = displayHint.contains("immersive") || body.exists{ b =>
+  override lazy val hideUi: Boolean = body.exists{ b =>
     Jsoup.parseBodyFragment(b).body().getElementsByClass("element-interactive").attr("data-interactive").contains("/visuals-blank-page/")
   }
+  override lazy val isImmersive: Boolean = displayHint.contains("immersive")
```
Comment (OliverJAsh) [inline common/app/model/content.scala]: Not sure whether this value belongs here on the Interactive class, or on the Content class.

**Comment by paperboyo** [comment]
[whispering] Fullscreen API...

File: common/app/views/fragments/headDefault.scala.html [hunk 2]
```diff
@@ -1,21 +1,23 @@
 @(content: model.Content, showBadge: Boolean = false)(implicit request: RequestHeader)
 
-<header class="content__head u-cf">
-    <div class="gs-container">
-        <div class="content__main-column">
+@if(!content.isImmersive) {
```
Comment (rich-nguyen) [inline common/app/views/fragments/headDefault.scala.html]: could move this test to interactiveBody.scala.html ?

**Comment by rich-nguyen** [comment]
:+1:

File: common/app/views/fragments/headDefault.scala.html (hunk 2 above)
Comment (rich-nguyen) [inline common/app/views/fragments/headDefault.scala.html]: (then `isImmersive` could be a val of the Interactive class)

**Comment by OliverJAsh** [comment]
After some discussion, it was decided that we should *hide everything except for the (slim) header on immersive, interactive content*. Updated. Here is a screenshot:

image

**Comment by paperboyo** [comment]
Never liked all these people sharing interactives... ;-) But what will they do now?

**Comment by OliverJAsh** [comment]
@paperboyo I guess it's up to the interactive team to decide if they want to add their own social buttons.

**Comment by paperboyo** [comment]
@OliverJAsh Will they be able to add the ones you're hiding or will they be adding square ones in different colour, translucent and in 3D?

**Comment by paperboyo** [comment]
Apologies for the playful tone - must be my coming holidays...

Serious point being - if these buttons will be disconnected from frontend code/design (I do hope I am wrong!), then they won't stay consistent even now. Let alone in the future. They will also likely change positions, sizes, won't include share counts and so on and so forth...

If the lack of screen real estate is to be 'blamed' for their disappearance, then [whispering] Fullscreen API would make everybody* happy ;-)

Regards
Mateusz

*apart from poor folks on Apple devices, but then, they are a dying breed...

**Comment by OliverJAsh** [comment]
@paperboyo That's a good point. We need to release this for a special interactive on Monday, but I will advocate this idea to the interactive team next week. Written down!

/cc @andymason

**Comment by paperboyo** [comment]
@OliverJAsh Thank you kindly and have a nice weekend, all!

**Pull request merged**